### PR TITLE
Font Library: refactor endpoint permissions

### LINKED ISSUE #54751: Font Library: fix endpoint permissions
It is necessary to fix the font library endpoints font permissions check.

## Check the right folder

We need to check for write permissions in the right folder:
Currently, `wp-content/upload/fonts/` is checked instead of `wp/content/fonts`.

## Endpoints

- ` GET /fonts/collections/`
- ` GET /fonts/collections/<id>/`
Should not check for write permissions on disk.
---

- `POST /fonts `
Should check for write permissions on disk only if any font families received have font faces defined with reference to files: `downloadFromUrl` or `uploadedFile`.
Should return HTTP error code 403 if there are no permissions.
---

- `DELETE /fonts`
Should not check write permissions. The font asset will be try to be removed and if it fails it will return error for that particular font face.

---

### PR DESCRIPTION
**Pull request by matiasbenedetto**
## What?
Font Library:
- Refactor endpoint permissions.
- Standardize the output of the install and uninstall endpoints.
- Display a success or error notice on the frontend for install and uninstall actions.

## Why?
- To have a standard output for the endpoints for success and error cases.

## How?
Refactoring the endpoint responses.

## Testing Instructions
- Use the font library to install local and Google fonts.
- Remove fonts
Check that the result is expected and you get the success or error notices.

Props to @pbking and @jffng for the big help implementing this :)

Closes: https://github.com/WordPress/gutenberg/issues/54751

### CODE REVIEW

File: lib/experimental/fonts/font-library/class-wp-rest-font-library-controller.php [hunk 1]
```diff
@@ -291,6 +291,22 @@ public function uninstall_schema() {
...
 
+		if ( empty( $fonts_to_install ) ) {
+			return new WP_Error(
+				'no_fonts_to_install',
+				__( 'No fonts to uninstall', 'gutenberg' ),
+				array( 'status' => 400 )
+			);
+		}
+
+		$needs_write_permission = $this->needs_write_permission( $fonts_to_install );
```
Comment (pbking) [inline lib/experimental/fonts/font-library/class-wp-rest-font-library-controller.php]: I didn't think that we planned to test the directory when uninstalling.
Can we remove this?

If not it looks like fonts_to_(un)install isn't being set.

**Review by jffng — APPROVED** [review]
This is working as expected for me: 

https://github.com/WordPress/gutenberg/assets/5375500/d8b76c1e-6949-4020-b5f4-5205c0f2c71a

I think the notices can be iterated on but it is good to have this as a starting point.

Also the tests are passing for me locally, so giving this the ✅

File: lib/experimental/fonts/font-library/class-wp-rest-font-library-controller.php (hunk 1 above)
Comment (matiasbenedetto) [reply lib/experimental/fonts/font-library/class-wp-rest-font-library-controller.php]: This was addressed in our pairing session. Thanks! Marking this as solved :heavy_check_mark:

**Review by pbking — APPROVED** [review]

**Comment by mikachan** [comment]
I just cherry-picked this PR to the commits-for-6.4-beta2 branch to get it included in the next release: fd167dbca5

**Pull request merged**